File: app/src/main/java/org/sil/storyproducer/controller/pager/PagerBaseActivity.java
package org.sil.storyproducer.controller.pager;

import android.os.Bundle;
import android.support.v4.view.ViewPager;

import org.sil.storyproducer.R;
import org.sil.storyproducer.controller.phase.PhaseBaseActivity;
import org.sil.storyproducer.model.Workspace;

/**
 * The activty that is the base of the paging views
 */
public class PagerBaseActivity extends PhaseBaseActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pager_base);

        // ViewPager and its adapters use support library
        // fragments, so use getSupportFragmentManager.
        PagerAdapter mPagerAdapter = new PagerAdapter(getSupportFragmentManager());
        ViewPager mViewPager = findViewById(R.id.pager);
        mViewPager.setAdapter(mPagerAdapter);
        //See if the slide number is valid for the new phase.  If not, set it to zero.
        if (!Workspace.INSTANCE.getActivePhase().checkValidDisplaySlideNum(
                Workspace.INSTANCE.getActiveSlideNum()))
            Workspace.INSTANCE.setActiveSlideNum(0);
        int slideNum = Workspace.INSTANCE.getActiveSlideNum();
        //If a slide other than 0 is chosen, use it.  This is to help recalling the last active slide.
        if(slideNum > 0) mViewPager.setCurrentItem(slideNum);

        mViewPager.addOnPageChangeListener(new CircularViewPagerHandler(mViewPager));       //sets the change listener to be the circular handler
    }

    @Override
    public void onPause() {
        super.onPause();
    }

}

File: app/src/main/java/org/sil/storyproducer/controller/pager/PagerBaseActivity.kt
package org.sil.storyproducer.controller.pager

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.sil.storyproducer.R
import org.sil.storyproducer.model.Workspace
import org.sil.storyproducer.model.PHASE_TYPE
import org.sil.storyproducer.model.PhaseType

class PagerBaseFragment : Fragment() {

    private var viewIsPrepared = false

    private lateinit var phaseType: PhaseType
    private lateinit var mPagerAdapter: PagerAdapter
    private lateinit var mViewPager: ViewPager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        phaseType = PhaseType.ofInt(arguments!!.getInt(PHASE_TYPE, 0))

        if (!phaseType.checkValidDisplaySlideNum(Workspace.activeStory.lastSlideNum)) {
            Workspace.activeStory.lastSlideNum = 0
        }

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        val rootView = inflater.inflate(R.layout.activity_pager_base, container, false)
        mPagerAdapter = PagerAdapter(childFragmentManager, phaseType)
        mViewPager = rootView.findViewById<ViewPager>(R.id.pager)
        mViewPager.adapter = mPagerAdapter
        mViewPager.currentItem = Workspace.activeStory.lastSlideNum
        mViewPager.addOnPageChangeListener(CircularViewPagerHandler(mViewPager))
        viewIsPrepared = true
        return rootView
    }

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)

        if (viewIsPrepared) {
            val itemIndex = Workspace.activeStory.lastSlideNum
            mViewPager.currentItem = itemIndex
            val page = mPagerAdapter.instantiateItem(mViewPager, itemIndex) as Fragment
            page.setUserVisibleHint(isVisibleToUser)
        }
    }
}
